ast/ast.go

```diff
@@ -1570,8 +1570,9 @@ func (n *SequenceNode) blockStyleString() string {
 		diffLength := len(splittedValues[0]) - len(trimmedFirstValue)
 		if len(splittedValues) > 1 && value.Type() == StringType || value.Type() == LiteralType {
 			// If multi-line string, the space characters for indent have already been added, so delete them.
+			prefix := space + "  "
```

Comment:
```suggestion
			firstContentLineSpaceNum := len(splittedValues[1]) - len(strings.TrimLeft(splittedValues[1], " "))
			prefix := strings.Repeat(" ", firstContentLineSpaceNum)
```

Reply:
I felt it would be better to calculate using the number of spaces in the first line

Reply:
That does not work if the first line is empty. I have added a test case with the first line empty. I've also discovered a separate issue https://github.com/goccy/go-yaml/issues/168.